### ISSUE TITLE
Help: Compile error

### ISSUE THREAD

**Issue by clint419** [foundry-rs/starknet-foundry#1080]
### Which component is your bug related to?

Forge

### Foundry Version

0.9.1

### What operating system are you using?

MacOS

### What system architecture are you using?

x86

### What happened

```shell
snforge test

Compiling sample v0.1.0 (/Users/suyanlong/github/zklink/stark/starknet-foundry/target/debug/sample/Scarb.toml)
    Finished release target(s) in 12 seconds
error: Inline macro `get_dep_component_mut` not found.
 --> /Users/suyanlong/Library/Caches/com.swmansion.scarb/registry/git/checkouts/cairo-contracts-9cboa8jg3jldq/4eafccf/src/access/accesscontrol/accesscontrol.cairo:185:38
            let mut src5_component = get_dep_component_mut!(ref self, SRC5);
                                     ^************************************^

error: Type annotations needed. Failed to infer ?0
 --> /Users/suyanlong/Library/Caches/com.swmansion.scarb/registry/git/checkouts/cairo-contracts-9cboa8jg3jldq/4eafccf/src/access/accesscontrol/accesscontrol.cairo:186:28
            src5_component.register_interface(interface::IACCESSCONTROL_ID);
                           ^****************^

error: Inline macro `get_dep_component_mut` not found.
 --> /Users/suyanlong/Library/Caches/com.swmansion.scarb/registry/git/checkouts/cairo-contracts-9cboa8jg3jldq/4eafccf/src/access/accesscontrol/accesscontrol.cairo:185:38
            let mut src5_component = get_dep_component_mut!(ref self, SRC5);
                                     ^************************************^

error: Type annotations needed. Failed to infer ?0
 --> /Users/suyanlong/Library/Caches/com.swmansion.scarb/registry/git/checkouts/cairo-contracts-9cboa8jg3jldq/4eafccf/src/access/accesscontrol/accesscontrol.cairo:186:28
            src5_component.register_interface(interface::IACCESSCONTROL_ID);
                           ^****************^

error: Inline macro `get_dep_component_mut` not found.
 --> /Users/suyanlong/Library/Caches/com.swmansion.scarb/registry/git/checkouts/cairo-contracts-9cboa8jg3jldq/4eafccf/src/account/account.cairo:192:38
            let mut src5_component = get_dep_component_mut!(ref self, SRC5);
                                     ^************************************^

error: Type annotations needed. Failed to infer ?0
 --> /Users/suyanlong/Library/Caches/com.swmansion.scarb/registry/git/checkouts/cairo-contracts-9cboa8jg3jldq/4eafccf/src/account/account.cairo:193:28
            src5_component.register_interface(interface::ISRC6_ID);
                           ^****************^

error: Inline macro `get_dep_component_mut` not found.
 --> /Users/suyanlong/Library/Caches/com.swmansion.scarb/registry/git/checkouts/cairo-contracts-9cboa8jg3jldq/4eafccf/src/account/account.cairo:192:38
            let mut src5_component = get_dep_component_mut!(ref self, SRC5);
```

Env:
- scarb version:  scarb 2.3.1 (0c8def3aa 2023-10-31)
- snforge 0.9.1

Scarb.toml:
```toml
[package]
name = "sample"
version = "0.1.0"

# See more keys and their definitions at https://docs.swmansion.com/scarb/docs/reference/manifest.html

[dependencies]
snforge_std = { git = "https://github.com/foundry-rs/starknet-foundry", tag = "v0.9.1" }
starknet = "2.3.1"
openzeppelin = { git = "https://github.com/OpenZeppelin/cairo-contracts.git", branch = "main" }


[[target.starknet-contract]]
casm = true
```

scarb build is succeed
scarb test is also succeed
but, snforge test fail

### Is there an existing issue for this?

- [X] I have searched the existing issues and verified no issue exits for this problem.

**Comment by zklinkJohn** [comment]
I also encountered the same problem.
I saw that the cairo version used by [cargo.toml](https://github.com/foundry-rs/starknet-foundry/blob/master/Cargo.toml) is still 2.3.0. I am not sure if this is the reason.

**Comment by piotmag769** [comment]
@zklinkJohn @clint419 Compiler version may be the reason for that, maybe this macro is not yet supported in 2.3.0. We will bump to 2.3.1 as I see some ppl need it

**Comment by Arcticae** [comment]
This macro is only available in the 2.4.0 versions, please consider downgrading your contracts/scarb until we are able to bump blockifier (which does not support 2.4.0 atm)

**Comment by Arcticae** [comment]
Should be resolved by #1087

**Comment by Arcticae** [comment]
#1087
